sanic/request.py

```diff
@@ -144,6 +145,9 @@ def __init__(
         self.transport = transport
         self.head = head
 
+        # enforce existing behavior prior to issue #2427
+        self.keep_blank_form_values: bool = False
```

Comment:
Should we also implement the same feature for query params?

We should think a little more on the API as well. While it absolutely should be controllable per `Request` instance, I can also see wanting to control this globally. Furthermore, if we implement for query params and form, should they have two different API values? Or, a unified API like this:

```python
class KeepBlank(Enum):
    NONE = ...
    FORM_ONLY = ...
    QUERY_ONLY = ...
    ALL = ...

keep_blank_values: bool | KeepBlank
```